libsupport/include/katana/CpuGpuSwitch.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace katana {
+
+//TODO (serge): change to a polymorphic allocator to switch between pinned and swappable memory
+enum class MemoryPinType { Swappable = 0, Pinned = 1 };
+
+#ifdef KATANA_ENABLE_GPU
+static constexpr const bool kGpuEnabled = true;
+static constexpr const MemoryPinType kUseMemoryPinType = MemoryPinType::Pinned;
+#else
+static constexpr const bool kGpuEnabled = false;
+static constexpr const MemoryPinType kUseMemoryPinType =
+    MemoryPinType::Swappable;
+#endif  // KATANA_ENABLE_GPU
```

Comment:
I still don't see why this needs to be in open. The open codebase doesn't support GPUs and these flags will need to be replaced by parameters or allocators soon anyway. So having them in open is a recipe for trouble since if anyone uses it, it's just more to fix as we refactor.

In the end, I'll leave it to you. But if there is no current need for this in open, I strongly recommend not including it.